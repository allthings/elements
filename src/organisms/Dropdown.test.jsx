import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider, ResourceProvider, Dropdown } from '../'

describe('Dropdown', () => {
  it('should render with placement bottom (default)', () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <ResourceProvider>
            <Dropdown
              initialSelectedItem={{
                label: 'dog',
                value: 'dog',
              }}
              label="some label"
              menuHeight={200}
              onSelect={() => {}}
              placeholder="select something..."
              clearable
              items={[
                {
                  label: 'dog',
                  value: 'dog',
                },
                {
                  label: 'cat',
                  value: 'cat',
                },
                {
                  label: 'apple',
                  value: 'apple',
                },
              ]}
            />
          </ResourceProvider>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render with placement top', () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <ResourceProvider>
            <Dropdown
              initialSelectedItem={{
                label: 'dog',
                value: 'dog',
              }}
              label="some label"
              menuHeight={200}
              onSelect={() => {}}
              placeholder="select something..."
              clearable
              items={[
                {
                  label: 'dog',
                  value: 'dog',
                },
                {
                  label: 'cat',
                  value: 'cat',
                },
                {
                  label: 'apple',
                  value: 'apple',
                },
              ]}
              placement="top"
            />
          </ResourceProvider>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
