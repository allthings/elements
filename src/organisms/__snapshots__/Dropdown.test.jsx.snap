// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render with placement bottom (default) 1`] = `
.css-0,
[data-css-0] {
  position: relative;
}

.css-1,
[data-css-1] {
  width: 100%;
  height: 50px;
  position: relative;
  background-color: #ffffff;
}

.css-2,
[data-css-2] {
  position: absolute;
  left: 15px;
  font-size: 10px;
  opacity: 0;
  transition: all .225s ease-out;
  -webkit-transition: all .225s ease-out;
  -moz-transition: all .225s ease-out;
}

.css-3,
[data-css-3] {
  align-items: stretch;
  border: none;
  box-shadow: 1px 1px 3px rgba(29, 29, 29, 0.125);
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  padding: 0;
  width: 100%;
  background-color: #ffffff;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-4,
[data-css-4] {
  width: 100%;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  cursor: pointer;
  padding-right: 30px;
  color: black;
}

.css-5,
[data-css-5] {
  cursor: pointer;
  height: 100%;
  width: 50px;
  padding: 16px 16px 16px 20px;
}

.css-6,
[data-css-6] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

.css-7,
[data-css-7] {
  display: inline;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-size: 13px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  box-sizing: border-box;
  height: 50px;
  width: 100%;
  padding: 0 15px;
  padding-top: 10px;
  transition: padding-top .225s ease-out;
  border: 0;
  -webkit-transition: padding-top .225s ease-out;
  -moz-transition: padding-top .225s ease-out;
}

.css-7:-webkit-autofill ~ .label,
[data-css-7]:-webkit-autofill ~ .label {
  opacity: 1 !important;
  top: 8px !important;
}

.css-7:-webkit-autofill ~ .checkmark,
[data-css-7]:-webkit-autofill ~ .checkmark {
  opacity: 1 !important;
  top: 8px;
}

.css-7:-webkit-autofill,
[data-css-7]:-webkit-autofill {
  padding-top: 10px !important;
}

.css-9,
[data-css-9] {
  position: absolute;
  top: 16px;
  right: 15px;
  pointer-events: none;
  transition: opacity .225s;
  opacity: 0;
  -webkit-transition: opacity .225s;
  -moz-transition: opacity .225s;
}

.css-10,
[data-css-10] {
  display: block;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-size: 10px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #626262;
}

.css-11,
[data-css-11] {
  position: absolute;
  top: 0;
  right: 0;
}

<div
  aria-expanded={false}
  aria-haspopup="listbox"
  aria-labelledby="downshift-0-label"
  aria-owns={null}
  data-css-3=""
  role="combobox"
>
  <div
    data-css-0=""
    data-css-6=""
  >
    <div
      data-css-6=""
      data-css-1=""
      onClick={[Function]}
    >
      <div
        data-css-0=""
        data-css-6=""
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <input
          aria-required={false}
          data-css-4=""
          data-css-7=""
          disabled={true}
          onChange={[Function]}
          placeholder="select something..."
          readOnly={true}
          required={false}
          type="text"
          value="dog"
        />
        <div
          className="label"
          data-css-6=""
          data-css-2=""
          style={
            Object {
              "opacity": 1,
              "top": 8,
            }
          }
        >
          <div
            data-css-6=""
            data-css-10=""
          >
            some label
             
          </div>
        </div>
        <div
          className="checkmark"
          data-css-9=""
          data-css-6=""
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
            data-css-6=""
            style={
              Object {
                "fill": "lightGrey",
                "height": 16,
                "stroke": false,
                "width": 16,
              }
            }
          />
        </div>
      </div>
      <div
        data-css-5=""
        data-css-11=""
        data-css-6=""
        onClick={[Function]}
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          data-css-6=""
          style={
            Object {
              "fill": false,
              "height": "10px",
              "stroke": "black",
              "width": "10px",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    data-css-0=""
    data-css-6=""
  />
</div>
`;

exports[`Dropdown should render with placement top 1`] = `
.css-0,
[data-css-0] {
  position: relative;
}

.css-1,
[data-css-1] {
  width: 100%;
  height: 50px;
  position: relative;
  background-color: #ffffff;
}

.css-2,
[data-css-2] {
  position: absolute;
  left: 15px;
  font-size: 10px;
  opacity: 0;
  transition: all .225s ease-out;
  -webkit-transition: all .225s ease-out;
  -moz-transition: all .225s ease-out;
}

.css-3,
[data-css-3] {
  align-items: stretch;
  border: none;
  box-shadow: 1px 1px 3px rgba(29, 29, 29, 0.125);
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  padding: 0;
  width: 100%;
  background-color: #ffffff;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-4,
[data-css-4] {
  width: 100%;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  cursor: pointer;
  padding-right: 30px;
  color: black;
}

.css-5,
[data-css-5] {
  cursor: pointer;
  height: 100%;
  width: 50px;
  padding: 16px 16px 16px 20px;
}

.css-6,
[data-css-6] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

.css-7,
[data-css-7] {
  display: inline;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-size: 13px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  box-sizing: border-box;
  height: 50px;
  width: 100%;
  padding: 0 15px;
  padding-top: 10px;
  transition: padding-top .225s ease-out;
  border: 0;
  -webkit-transition: padding-top .225s ease-out;
  -moz-transition: padding-top .225s ease-out;
}

.css-7:-webkit-autofill ~ .label,
[data-css-7]:-webkit-autofill ~ .label {
  opacity: 1 !important;
  top: 8px !important;
}

.css-7:-webkit-autofill ~ .checkmark,
[data-css-7]:-webkit-autofill ~ .checkmark {
  opacity: 1 !important;
  top: 8px;
}

.css-7:-webkit-autofill,
[data-css-7]:-webkit-autofill {
  padding-top: 10px !important;
}

.css-9,
[data-css-9] {
  position: absolute;
  top: 16px;
  right: 15px;
  pointer-events: none;
  transition: opacity .225s;
  opacity: 0;
  -webkit-transition: opacity .225s;
  -moz-transition: opacity .225s;
}

.css-10,
[data-css-10] {
  display: block;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-size: 10px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #626262;
}

.css-11,
[data-css-11] {
  position: absolute;
  top: 0;
  right: 0;
}

<div
  aria-expanded={false}
  aria-haspopup="listbox"
  aria-labelledby="downshift-1-label"
  aria-owns={null}
  data-css-3=""
  role="combobox"
>
  <div
    data-css-0=""
    data-css-6=""
  >
    <div
      data-css-6=""
      data-css-1=""
      onClick={[Function]}
    >
      <div
        data-css-0=""
        data-css-6=""
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <input
          aria-required={false}
          data-css-4=""
          data-css-7=""
          disabled={true}
          onChange={[Function]}
          placeholder="select something..."
          readOnly={true}
          required={false}
          type="text"
          value="dog"
        />
        <div
          className="label"
          data-css-6=""
          data-css-2=""
          style={
            Object {
              "opacity": 1,
              "top": 8,
            }
          }
        >
          <div
            data-css-6=""
            data-css-10=""
          >
            some label
             
          </div>
        </div>
        <div
          className="checkmark"
          data-css-9=""
          data-css-6=""
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            data-css-6=""
            style={
              Object {
                "fill": "lightGrey",
                "height": 16,
                "stroke": false,
                "width": 16,
              }
            }
          />
        </div>
      </div>
      <div
        data-css-5=""
        data-css-11=""
        data-css-6=""
        onClick={[Function]}
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
          data-css-6=""
          style={
            Object {
              "fill": false,
              "height": "10px",
              "stroke": "black",
              "width": "10px",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    data-css-0=""
    data-css-6=""
  />
</div>
`;
